x/bank/keeper/keeper.go

```diff
@@ -24,8 +24,12 @@ type Keeper interface {
 	InitGenesis(sdk.Context, *types.GenesisState)
 	ExportGenesis(sdk.Context) *types.GenesisState
 
-	GetSupply(ctx sdk.Context) exported.SupplyI
-	SetSupply(ctx sdk.Context, supply exported.SupplyI)
+	GetSupply(ctx sdk.Context, denom string) sdk.Coin
+	GetTotalSupply(ctx sdk.Context) sdk.Coins
```

Comment:
This should really include some sort of pagination or just be removed...

Reply:
Agree, this is especially relevant in the context of IBC

Reply:
Can we make sure there's a follow up issue to address this?